File: P5 2024-2/Paradigmas de Programacao/ListaExercicio01/Golang/Escola.go
package main

import "fmt"

type Professor struct {
	nome    string
	Escolas []*Escola
}

type Escola struct {
	nome        string
	Professores []*Professor
}

func (e *Escola) contrataProfessor(p *Professor) {
	e.Professores = append(e.Professores, p)
	p.Escolas = append(p.Escolas, e)
}

func (e *Escola) listaProfessores() {
	fmt.Println("Listando professores da escola ", e.nome)
	for _, p := range e.Professores {
		fmt.Println(p.nome)
	}
}

func (p *Professor) listaEscolas() {
	fmt.Println("Listando escolas do professor ", p.nome)
	for _, e := range p.Escolas {
		fmt.Println(e.nome)
	}
}

File: P5 2024-2/Paradigmas de Programacao/ListaExercicio01/Python/Escola.py
class Professor:
    def __init__(self, nome):
        self.nome = nome
        self.Escolas = []

    def listaEscolas(self):
        print("Listando escolas do professor", self.nome)
        for e in self.Escolas:
                print(e.nome)

class Escola:
    def __init__(self, nome):
        self.nome = nome
        self.Professores = []

    def contrataProfessor(self, professor):
        self.Professores.append(professor)
        professor.Escolas.append(self)

    def listaProfessores(self):
        print("Listando Professores da escola", self.nome)
        for p in self.Professores:
            print(p.nome)